### PR TITLE
Adding the resources for ibm scc templates and ibm scc rules

### PR DESCRIPTION
**Pull request by tyao117**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

**Comment by hkantare** [comment]
Add respective doc files for each resource
Fix the conflict in ibm/conns/config.go
Attach the results of test case executions..

File: ibm/service/scc/resource_ibm_scc_rule.go [hunk 1]
```diff
@@ -0,0 +1,1079 @@
...
+func ResourceIBMSccRule() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceIBMSccRuleCreate,
+		ReadContext:   resourceIBMSccRuleRead,
+		UpdateContext: resourceIBMSccRuleUpdate,
+		DeleteContext: resourceIBMSccRuleDelete,
+		Importer:      &schema.ResourceImporter{},
+
+		Schema: map[string]*schema.Schema{
+			"account_id": &schema.Schema{
```
[the remark this answers is not in the document]
Comment (tyao117) [reply ibm/service/scc/resource_ibm_scc_rule.go:38]: @hkantare , we may not want to target the same account. we should allow the ability to target a different account.

**Comment by tyao117** [comment]
```shell
(.direnv)➜  terraform-provider-ibm git:(tyao117_update_rules_templates) ✗ make test                                                                                                               ==> Checking that code complies with gofmt requirements...                                                                                                                                        
go test -i $(go list ./... |grep -v 'vendor') || exit 1                                                                                                                                           
go test: -i flag is deprecated                                                                                                                                                                    
go: downloading golang.org/x/term v0.0.0-20210927222741-03fcf44c2211                                                                                                                              echo $(go list ./... |grep -v 'vendor') | \                                                                                                                                                                       xargs -t -n4 go test  -timeout=30s -parallel=4                                                                                                                                    go test -timeout=30s -parallel=4 github.com/IBM-Cloud/terraform-provider-ibm github.com/IBM-Cloud/terraform-provider-ibm/ibm/acctest github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns github.com/IBM-Cloud/terraform-provider-ibm/ibm/flex                                                                                                                                                     
?       github.com/IBM-Cloud/terraform-provider-ibm     [no test files]                                                                                                                           ?       github.com/IBM-Cloud/terraform-provider-ibm/ibm/acctest [no test files]                                                                                                                   
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns   0.546s                                                                                                                            
?       github.com/IBM-Cloud/terraform-provider-ibm/ibm/flex    [no test files]                                                                                                                   
go test -timeout=30s -parallel=4 github.com/IBM-Cloud/terraform-provider-ibm/ibm/provider github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/apigateway github.com/IBM-Cloud/terraform-provid
er-ibm/ibm/service/appconfiguration github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/appid                                                                                                 
?       github.com/IBM-Cloud/terraform-provider-ibm/ibm/provider        [no test files]                                                                                                           
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/apigateway      1.128s                                                                                                            ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/appconfiguration        2.612s                                                                                                    ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/appid   1.524s                                                                                                                    go test -timeout=30s -parallel=4 github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/atracker github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/catalogmanagement github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/certificatemanager github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/cis                                                                                  ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/atracker        1.736s                                                                                                            
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/catalogmanagement       2.289s                                                                                                    
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/certificatemanager      1.359s                                                                                                    
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/cis     2.958s                                                                                                                    
go test -timeout=30s -parallel=4 github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/classicinfrastructure github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/cloudant github.com/IBM-Clo
ud/terraform-provider-ibm/ibm/service/cloudfoundry github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/cloudshell                                                                             
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/classicinfrastructure   3.829s                                                                                               
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/cloudant        1.321s                                                                                                            
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/cloudfoundry    1.721s                                                                                                            
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/cloudshell      1.009s
go test -timeout=30s -parallel=4 github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/contextbasedrestrictions github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/cos github.com/IBM-Cloud
/terraform-provider-ibm/ibm/service/database github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/directlink
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/contextbasedrestrictions        0.924s
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/cos     1.974s
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/database        2.775s
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/directlink      1.440s
go test -timeout=30s -parallel=4 github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/dnsservices github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/enterprise github.com/IBM-Cloud/terra
form-provider-ibm/ibm/service/eventnotification github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/eventstreams
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/dnsservices     2.404s
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/enterprise      1.015s
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/eventnotification       2.808s
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/eventstreams    1.608s
go test -timeout=30s -parallel=4 github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/functions github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/globaltagging github.com/IBM-Cloud/terr
aform-provider-ibm/ibm/service/hpcs github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/iamaccessgroup
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/functions       3.147s
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/globaltagging   1.646s
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/hpcs    1.894s
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/iamaccessgroup  2.572s
go test -timeout=30s -parallel=4 github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/iamidentity github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/iampolicy github.com/IBM-Cloud/terraf
orm-provider-ibm/ibm/service/kms github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/kubernetes
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/iamidentity     2.120s
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/iampolicy       2.649s
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/kms     1.608s
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/kubernetes      3.628s
go test -timeout=30s -parallel=4 github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/power github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/pushnotification github.com/IBM-Cloud/terra
form-provider-ibm/ibm/service/registry github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/resourcecontroller
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/power   3.889s
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/pushnotification        3.110s
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/registry        2.255s
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/resourcecontroller      1.332s
go test -timeout=30s -parallel=4 github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/resourcemanager github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/satellite github.com/IBM-Cloud/te
rraform-provider-ibm/ibm/service/scc github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/schematics
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/resourcemanager 1.382s
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/satellite       3.126s
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/scc     3.587s
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/schematics      2.091s
go test -timeout=30s -parallel=4 github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/secretsmanager github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/transitgateway github.com/IBM-Clou
d/terraform-provider-ibm/ibm/service/vpc github.com/IBM-Cloud/terraform-provider-ibm/ibm/validate 
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/secretsmanager  1.302s
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/transitgateway  1.978s
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc     2.271s
?       github.com/IBM-Cloud/terraform-provider-ibm/ibm/validate        [no test files]
go test -timeout=30s -parallel=4 github.com/IBM-Cloud/terraform-provider-ibm/version
?       github.com/IBM-Cloud/terraform-provider-ibm/version     [no test files]
```

**Comment by tyao117** [comment]
here is the test results from the make test:
```shell
(.direnv)➜  terraform-provider-ibm git:(tyao117_update_rules_templates) ✗ make testacc TEST=./ibm/service/scc TESTARGS='-run=TestAccIBMSccRule'                      
==> Checking that code complies with gofmt requirements...                                                                                                                                                                                      
TF_ACC=1 go test ./ibm/service/scc -v -run=TestAccIBMSccRule -timeout 700m
...
=== RUN   TestAccIBMSccRuleAttachmentBasic                                                                                                                                                                                                      
--- PASS: TestAccIBMSccRuleAttachmentBasic (9.71s)                                                                                                                                                                                              
=== RUN   TestAccIBMSccRuleBasic                                                                                                                                                                                                                
--- PASS: TestAccIBMSccRuleBasic (6.96s)                                                                                                                                                                                                        
PASS                                                                                                                                                                                                                                            
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/scc     17.925s
```
this is for the templates:
```shell
make testacc TEST=./ibm/service/scc TESTARGS='-run=TestAccIBMSccTemplate'
==> Checking that code complies with gofmt requirements...                                                                                                                                                                                      
TF_ACC=1 go test ./ibm/service/scc -v -run=TestAccIBMSccTemplate -timeout 700m                                                                        
...
=== RUN   TestAccIBMSccTemplateAttachmentBasic
--- PASS: TestAccIBMSccTemplateAttachmentBasic (8.88s)
=== RUN   TestAccIBMSccTemplateBasic
--- PASS: TestAccIBMSccTemplateBasic (7.13s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/scc     16.958s
```

File: website/docs/r/scc_configuration_rule.html.markdown [hunk 2]
```diff
@@ -0,0 +1,228 @@
+---
+layout: "ibm"
+page_title: "IBM : ibm_scc_rule"
+description: |-
+  Manages scc_rule.
+subcategory: "Configuration Governance API"
```
[the remark this answers is not in the document]
Comment (hkantare) [reply website/docs/r/scc_configuration_rule.html.markdown:6]: Once above change is done we are good for merge for this PR

[the remark this answers is not in the document]
Comment (tyao117) [reply website/docs/r/scc_configuration_rule.html.markdown]: Change is now pushed. Going to ask for the review again.

**Pull request merged**